### PR TITLE
motion correction models

### PR DESCRIPTION
**Pull request by luiztauffer**

### CODE REVIEW

File: src/spikeinterface_pipelines/preprocessing/preprocessing.py [hunk 1]
```diff
@@ -105,7 +105,15 @@ def preprocess(
         logger.info(f"[Preprocessing] \tComputing motion correction with preset: {preset}")
         motion_folder = results_folder / "motion_correction"
         recording_corrected = spre.correct_motion(
-            recording_processed, preset=preset, folder=motion_folder, verbose=False
+            recording_processed,
+            preset=preset,
+            folder=motion_folder,
+            verbose=False,
+            detect_kwargs=preprocessing_params.motion_correction.motion_kwargs.detect_kwargs.model_dump(),
+            select_kwargs=dict(),
```
Comment (alejoe91) [inline src/spikeinterface_pipelines/preprocessing/preprocessing.py]:
```suggestion
```
This is not used and there is no model for it, so I would avoid passing it

File: src/spikeinterface_pipelines/preprocessing/preprocessing.py [hunk 2]
```diff
@@ -105,7 +105,15 @@ def preprocess(
         logger.info(f"[Preprocessing] \tComputing motion correction with preset: {preset}")
         motion_folder = results_folder / "motion_correction"
```
Comment (alejoe91) [inline src/spikeinterface_pipelines/preprocessing/preprocessing.py:112]: Should we add a check that the motion kwargs correspond to the right motion preset here?

File: src/spikeinterface_pipelines/preprocessing/params.py [hunk 3]
```diff
@@ -41,12 +41,125 @@ class HighpassSpatialFilter(BaseModel):
     highpass_butter_wn: float = Field(default=0.01, description="Natural frequency for the Butterworth filter")
 
 
+# Motion correction ---------------------------------------------------------------
+class MCDetectKwargs(BaseModel):
+    method: str = Field(default="locally_exclusive", description="")
+    peak_sign: str = Field(default="neg", description="")
+    detect_threshold: float = Field(default=8.0, description="")
+    exclude_sweep_ms: float = Field(default=0.1, description="")
+    radius_um: float = Field(default=50.0, description="")
```
Comment (alejoe91) [inline src/spikeinterface_pipelines/preprocessing/params.py]:
```suggestion
    method: str = Field(default="locally_exclusive", description="The method for peak detection.")
    peak_sign: Literal["pos", "neg", "both] = Field(default="neg", description="The peak sign to detect peaks.")
    detect_threshold: float = Field(default=8.0, description="The detection threshold in MAD units.")
    exclude_sweep_ms: float = Field(default=0.1, description="The time sweep to exclude for time de-duplication.")
    radius_um: float = Field(default=50.0, description="The radius in um for channel de-duplication.")
```

File: src/spikeinterface_pipelines/preprocessing/params.py [hunk 4]
```diff
@@ -41,12 +41,125 @@ class HighpassSpatialFilter(BaseModel):
...
+
+
+class MCLocalizeCenterOfMass(BaseModel):
+    radius_um: float = Field(default=75.0, description="Radius in um for channel sparsity.")
+    feature: str = Field(default="ptp", description="'ptp', 'mean', 'energy' or 'peak_voltage'. Feature to consider for computation")
+
+
+class MCLocalizeMonopolarTriangulation(BaseModel):
+    radius_um: float = Field(default=75.0, description="For channel sparsity.")
```
Comment (alejoe91) [inline src/spikeinterface_pipelines/preprocessing/params.py]:
```suggestion
    radius_um: float = Field(default=75.0, description="Radius in um for channel sparsity.")
```

File: src/spikeinterface_pipelines/postprocessing/params.py [hunk 5]
```diff
@@ -94,9 +94,10 @@ class QMParams(BaseModel):
     nn_noise_overlap: NNIsolation = Field(default=NNIsolation(), description="Nearest neighbor noise overlap.")
 
 
+# TODO - fill in metric_names defauklt value with all the metric names
 class QualityMetrics(BaseModel):
     qm_params: QMParams = Field(default=QMParams(), description="Quality metric parameters.")
-    metric_names: List[str] = Field(default=None, description="List of metric names to compute.")
+    metric_names: List[str] = Field(default=[], description="List of metric names to compute.")
```
Comment (alejoe91) [inline src/spikeinterface_pipelines/postprocessing/params.py]:
```suggestion
    metric_names: Optional[List[str]] = Field(default=None, description="List of metric names to compute. If None, all available metrics are computed.")
```

**Review by alejoe91 — CHANGES_REQUESTED** [review]
Thansk @luiztauffer 

I have some suggestions and filled in some descriptions ;)

File: src/spikeinterface_pipelines/preprocessing/params.py [hunk 6]
```diff
@@ -41,12 +41,125 @@ class HighpassSpatialFilter(BaseModel):
...
+
+class MCPreset(str, Enum):
+    nonrigid_accurate = "nonrigid_accurate"
+    rigid_fast = "rigid_fast"
+    kilosort_like = "kilosort_like"
+
+
 class MotionCorrection(BaseModel):
     compute: bool = Field(default=True, description="Whether to compute motion correction")
     apply: bool = Field(default=False, description="Whether to apply motion correction")
```
Comment (alejoe91) [inline src/spikeinterface_pipelines/preprocessing/params.py]:
```suggestion
    strategy: Literal["skip", "compute", "apply"] = Field(default="compute", description="What strategy to use for motion correction")
```
What do you think about this? This is what we ended up using in our pipelines ;)

Comment (luiztauffer) [reply src/spikeinterface_pipelines/preprocessing/params.py]: @alejoe91 what is the difference between compute and apply? Is there any case where we would like to compute but to not apply?

File: src/spikeinterface_pipelines/preprocessing/preprocessing.py (hunk 2 above)
Comment (luiztauffer) [reply src/spikeinterface_pipelines/preprocessing/preprocessing.py:112]: done!

File: src/spikeinterface_pipelines/preprocessing/params.py (hunk 6 above)
Comment (alejoe91) [reply src/spikeinterface_pipelines/preprocessing/params.py]: yes, you might want to leave motion correction to KS, but still compute it externally for visualization or comparison

Comment (luiztauffer) [reply src/spikeinterface_pipelines/preprocessing/params.py]: sounds good then

**Comment by luiztauffer** [comment]
@alejoe91 good to merge?

**Pull request merged**